### ISSUE TITLE
Lint/NumericOperationWithConstantResult incorrectly handles i % 1

### ISSUE THREAD

**Issue by modosc** [rubocop/rubocop#13532]
--------

## Expected behavior
input: 
```ruby
if i % 1 == 0
```
output:
```ruby
if i % 1 == 0
```
## Actual behavior
input: 
```ruby
if i % 1 == 0
```
output:
```ruby
if 0 == 0
```

## Steps to reproduce the problem

`i % 1 == 0` is a valid way to determine if a numeric value has a fractional value:
```ruby
> 1 % 1 == 0
=> true
> 1.0 % 1 == 0
=> true
> 1.1 % 1 == 0
=> false
```

but the autofix from `Lint/NumericOperationWithConstantResult` breaks this. 
```
$ cat foo.rb
if i % 1 == 0

end
$ rubocop -A foo.rb
Inspecting 1 file
W

Offenses:

foo.rb:3:4: W: Lint/BinaryOperatorWithIdenticalOperands: Binary operator == has identical operands.
if 0 == 0
   ^^^^^^

1 file inspected, 1 offense detected

$ cat foo.rb
if 0 == 0

end
```

## RuboCop version

```
1.69.0 (using Parser 3.3.6.0, rubocop-ast 1.36.2, analyzing as Ruby 3.3, running on ruby 3.3.5) [arm64-darwin24]
  - rubocop-factory_bot 2.26.1
  - rubocop-performance 1.23.0
  - rubocop-rails 2.27.0
  - rubocop-rspec 3.2.0
  - rubocop-rspec_rails 2.30.0```

**Comment by viralpraxis** [comment]
I agree this is a bug. Cop description says

> Additionally, a variable modulo 0 or itself will always return 0.

(I believe "a variable modulo 0" was meant)

which is simply not true.

**Comment by koic** [comment]
This issue was resolved by #13577.
